### DISCUSSION TITLE
Vue3

### DISCUSSION THREAD

**Discussion by akorajac** [ghettovoice/vuelayers#445] [Ideas]
Great project, are there any plans to make it compatible with Vue3 in the future?

**Comment by matthewpavkov** [comment]
I'm also wondering the same. We're thinking about converting this to Vue3 ourselves, but we haven't decided for sure just yet.

**Comment by mibzman** [comment]
To my knowledge there are these issues relating to vue 3 support: #377 and #194

This other project exists: https://github.com/MelihAltintas/vue3-openlayers, so we're using that in the the meantime.

@ghettovoice, I'm also interested in contributing to porting this to vue 3, **what would be the major hurdles to upgrading**?  

I made an attempt to run this in vue 3, but I didn't really get any identifiable errors I could fix, it just didn't load.  I'd be happy to spend some time trying to upgrade sometime later this year if you could outline a way forward.

**Reply by akorajac** [reply]
yeah I wrote this before stumbling on Melih's vue3 project, which is very good and he is currently working on it and adding more stuff. 👍

**Reply by MelihAltintas** [reply]
Thanks for your comment :) I've been using vuelayers for a long time too. However, due to the new project, it was necessary to switch to vue3. I would like to contribute to Vuelayers, but many things used in vuelayers are removed in vue 3.x :( so to create a new repo is easier then maintain the vuelayers because of the removed features.

**Comment by ghettovoice** [comment]
Hi there!
Yeh , I have plans to upgrade to vue3, but for the time being I'm still busy on other projects. 
Actually this project became a side project, and now I don't have enough time to invest so much time as before into it. 

@mibzman it would be great if you want to contribute. This project need more active contributors 👍 
Frankly speaking I tried vue3 only once, just to check how it is working. Don't dig into details, especially how to migrate plugins. I guess there can be some incompatibles in the component lifetime, VueLayers create/mount OpenLayers components asyncheronously via Vue hooks and events. And sure^ if also migrate source code to Typescript - it is just a very big work...
